templates/hooks/cache_subnet_group/sdk_read_many_post_set_output.go.tpl

```diff
@@ -0,0 +1,6 @@
+
+        for _, subnetIdIter := range ko.Status.Subnets {
+                if subnetIdIter.SubnetIdentifier != nil {
+                        ko.Spec.SubnetIDs = append(ko.Spec.SubnetIDs, subnetIdIter.SubnetIdentifier)
+                }
+        }
```

Comment:
I think we missed that we might be duplicating subnets here (if a user already provides subnets they will be copied in sdk.go L87). Ideally, we should init a new one:
```go

        subnets := make([]*svcapitypes.Subnet, 0, len(resp.Subnets))
        for _, subnetIdIter := range ko.Status.Subnets {
                if subnetIdIter.SubnetIdentifier != nil {
                        subnets = append(subnets, subnetIdIter.SubnetIdentifier)
                }
        }
        ko.Spec.Subnets = subnets
```